src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/fastSetupCreator.m

```diff
@@ -38,18 +38,19 @@
     exch = {};
 end
 for j = 1:size(models, 1)
-    model = models{j, 1};
-    model = findSExRxnInd(model);
-    mod_exch=model.rxns(model.ExchRxnBool);
-    mod_exch=strrep(mod_exch, 'EX_', '');
-    mod_exch=strrep(mod_exch, '(e)', '[e]');
-    exch = union(exch,mod_exch);
+     model = models{j, 1};
+     exch = union(exch, model.mets(find(sum(model.S(:, strncmp('EX_', model.rxns, 3)), 2) ~= 0)));
+%      model = findSExRxnInd(model);
```

Comment:
I see what went wrong. you essentially got the biomass reaction in as well and added reactions instead of metabolites.
This should do what you want:
```
if ~isempty(host)  % Get list of all exchanged metabolites
    exStruct = findSExRxnInd(host);
    exch = findMetsFromRxns(host,host.rxns(exStruct.ExchRxnBool & ~exStruct.biomassBool));
else
    exch = {};
end
for j = 1:size(models, 1)
    model = models{j, 1};
    exStruct = findSExRxnInd(model);
    new_exch = findMetsFromRxns(model,model.rxns(exStruct.ExchRxnBool & ~exStruct.biomassBool));
    exch = union(exch,new_exch);
end
```

Reply:
Yes, the problem seemed to be that, enabling the function back and changing the call as you proposed worked....
Thanks

Reply:
@laurentheirendt